Extension/src/LanguageServer/lmTool.ts

```diff
@@ -44,6 +48,129 @@ const knownValues: { [Property in keyof ChatContextResult]?: { [id: string]: str
+                break;
+            case GCC:
+                filterMap = context.flags.copilotcppGccCompilerArgumentFilter ? JSON.parse(context.flags.copilotcppGccCompilerArgumentFilter as string ?? '{}') : undefined;
+                break;
+        }
+    }
+    catch {
+        // Intentionally swallow any exception.
+    }
+    return filterMap;
+}
+
+function filterComplierArguments(compiler: string, compilerArguments: string[], context: { flags: Record<string, unknown> }): string[] {
+    const filterMap = getCompilerArgumentFilterMap(compiler, context);
+    if (filterMap === undefined) {
+        return [];
+    }
+
+    const filteredArguments: string[] = [];
+    for (const key in filterMap) {
+        if (!key) {
+            continue;
+        }
+        const filter = new RegExp(key);
+        const filtered = compilerArguments.filter(arg => filter.test(arg));
+        filteredArguments.push(...filtered);
+    }
+    return filteredArguments;
+}
+
+export async function getProjectContext(uri: vscode.Uri, context: { flags: Record<string, unknown> }, token: vscode.CancellationToken): Promise<ProjectContext | undefined> {
+    const telemetryProperties: Record<string, string> = {};
+    try {
+        const projectContext = await checkTime<ProjectContextResult | undefined>(async () => await getClients()?.ActiveClient?.getProjectContext(uri, token) ?? undefined);
+        telemetryProperties["time"] = projectContext.time.toFixed(1);
+        if (!projectContext.result) {
+            return undefined;
+        }
+
+        formatChatContext(projectContext.result);
+
+        const result: ProjectContext = {
+            language: projectContext.result.language,
+            standardVersion: projectContext.result.standardVersion,
+            compiler: projectContext.result.compiler,
+            targetPlatform: projectContext.result.targetPlatform,
+            targetArchitecture: projectContext.result.targetArchitecture,
+            compilerArguments: []
+        };
+
+        if (projectContext.result.language) {
+            telemetryProperties["language"] = projectContext.result.language;
+        }
+        if (projectContext.result.compiler) {
+            telemetryProperties["compiler"] = projectContext.result.compiler;
+        }
+        if (projectContext.result.standardVersion) {
+            telemetryProperties["standardVersion"] = projectContext.result.standardVersion;
+        }
+        if (projectContext.result.targetPlatform) {
+            telemetryProperties["targetPlatform"] = projectContext.result.targetPlatform;
+        }
+        if (projectContext.result.targetArchitecture) {
+            telemetryProperties["targetArchitecture"] = projectContext.result.targetArchitecture;
+        }
+        telemetryProperties["compilerArgumentCount"] = projectContext.result.fileContext.compilerArguments.length.toString();

```

Comment:
Another property that should probably be a "metric"

Reply:
Does it matter if one string is sent instead of a number? I understand that metrics would make sense if you're sending a lot of properties with number values, but is there a reason to do this for 1 property that I'm not aware of?

Reply:
You can always convert it in Kusto, but when you're aggregating a lot of data, it makes more sense for it to be stored as the correct type. It's not the end of the world if you send it as a string, but we should try to be consisent and send numbers as numbers.

Reply:
Fixed, thanks.